app/app.go

```diff
@@ -76,22 +79,34 @@ type App interface {
 
 // app is an app object
 type app struct {
-	store   store.DataStore
-	hub     iothub.Client
-	wf      workflows.Client
-	devauth devauth.Client
+	store         store.DataStore
+	iothubClient  iothub.Client
+	iotcoreClient iotcore.Client
+	wf            workflows.Client
+	devauth       devauth.Client
 }
 
 // NewApp initialize a new iot-manager App
-func New(ds store.DataStore, hub iothub.Client, wf workflows.Client, da devauth.Client) App {
+func New(ds store.DataStore, wf workflows.Client, da devauth.Client) *app {
```

Comment:
why are we returning struct not the interface?
because the new With* methods are not part of the App interface.
why arent they?